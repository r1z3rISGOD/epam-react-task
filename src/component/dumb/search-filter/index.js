import {SearchFilter} from "./search-filter"
export {SearchFilter}
