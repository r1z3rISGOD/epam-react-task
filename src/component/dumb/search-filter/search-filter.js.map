{"version":3,"file":"search-filter.js","sourceRoot":"","sources":["search-filter.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,gCAA6B;AAOtB,IAAM,YAAY,GAAiC,UAAC,EAAuB;QAArB,UAAU,gBAAA,EAAE,OAAO,aAAA;IAC9E,IAAI,OAAO,KAAK,OAAO,EAAE;QACvB,OAAO,CACC,6BAAK,SAAS,EAAC,eAAe;YAC1B,8BAAM,SAAS,EAAC,qBAAqB,gBAAiB;YACtD,gCAAQ,SAAS,EAAC,oDAAoD,YAAe;YACrF,gCAAQ,OAAO,EAAE;oBACf,UAAU,CAAC,QAAQ,CAAC,CAAA;gBACtB,CAAC,EAAE,KAAK,EAAC,QAAQ,EAAC,SAAS,EAAC,2BAA2B,YAAe,CACpE,CACb,CAAA;KACF;IAAC,IAAI,OAAO,KAAK,QAAQ,EAAE;QAC1B,OAAO,CACC,6BAAK,SAAS,EAAC,eAAe;YAC1B,8BAAM,SAAS,EAAC,qBAAqB,gBAAiB;YACtD,gCAAQ,OAAO,EAAE;oBACf,UAAU,CAAC,OAAO,CAAC,CAAA;gBACrB,CAAC,EAAE,KAAK,EAAC,OAAO,EAAC,SAAS,EAAC,2BAA2B,YAAe;YACrE,gCAAQ,SAAS,EAAC,oDAAoD,YAAe,CACnF,CACb,CAAA;KACF;AACH,CAAC,CAAA;AAtBY,QAAA,YAAY,gBAsBxB"}