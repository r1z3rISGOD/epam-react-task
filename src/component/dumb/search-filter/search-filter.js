"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchFilter = void 0;
var React = require("react");
require("./search-filter.scss");
var SearchFilter = function (_a) {
    var setSetting = _a.setSetting, setting = _a.setting;
    if (setting === 'title') {
        return (React.createElement("div", { className: "search-filter" },
            React.createElement("span", { className: "search-filter__text" }, "SEARCH BY"),
            React.createElement("button", { onClick: function (e) {
                    setSetting(e.target.value);
                }, value: "title", className: "search-filter__button search-filter__button-active" }, "TITLE"),
            React.createElement("button", { onClick: function (e) {
                    setSetting(e.target.value);
                }, value: "genres", className: "search-filter__button" }, "GENRE")));
    }
    if (setting === 'genres') {
        return (React.createElement("div", { className: "search-filter" },
            React.createElement("span", { className: "search-filter__text" }, "SEARCH BY"),
            React.createElement("button", { onClick: function (e) {
                    setSetting(e.target.value);
                }, value: "title", className: "search-filter__button " }, "TITLE"),
            React.createElement("button", { onClick: function (e) {
                    setSetting(e.target.value);
                }, value: "genre", className: "search-filter__button search-filter__button-active" }, "GENRE")));
    }
};
exports.SearchFilter = SearchFilter;
//# sourceMappingURL=search-filter.js.map