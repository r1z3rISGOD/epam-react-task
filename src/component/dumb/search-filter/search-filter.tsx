import * as React from 'react'
import './search-filter.scss'

interface SearchFilterProps {
    setSetting: any
    setting: string
}

export const SearchFilter : React.FC<SearchFilterProps> = ({ setSetting, setting }) => {
  if (setting === 'title') {
    return (
            <div className="search-filter">
                <span className="search-filter__text">SEARCH BY</span>
                <button className="search-filter__button search-filter__button-active">TITLE</button>
                <button onClick={() => {
                  setSetting('genres')
                }} value="genres" className="search-filter__button btn">GENRE</button>
            </div>
    )
  } if (setting === 'genres') {
    return (
            <div className="search-filter">
                <span className="search-filter__text">SEARCH BY</span>
                <button onClick={() => {
                  setSetting('title')
                }} value="title" className="search-filter__button btn">TITLE</button>
                <button className="search-filter__button search-filter__button-active">GENRE</button>
            </div>
    )
  }
}
