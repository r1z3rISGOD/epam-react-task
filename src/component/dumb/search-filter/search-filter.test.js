import { SearchFilter } from './search-filter.tsx'
import React from 'react'

function f () {
  return 1
}

const setUp = (setting) => shallow(<SearchFilter setSetting={f} setting={setting}/>)

describe('search filter tests', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })

  it('search filter test 1', () => {
    const component = setUp('title')
    expect(component).toMatchSnapshot()
    const btn = component.find('.btn')
    btn.simulate('click')
  })
  it('search filter test 2', () => {
    const component = setUp('genres')
    expect(component).toMatchSnapshot()
    const btn = component.find('.btn')
    btn.simulate('click')
  })
})
