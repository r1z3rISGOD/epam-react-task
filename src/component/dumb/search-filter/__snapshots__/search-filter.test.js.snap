// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search filter tests search filter test 1 1`] = `
<div
  className="search-filter"
>
  <span
    className="search-filter__text"
  >
    SEARCH BY
  </span>
  <button
    className="search-filter__button search-filter__button-active"
  >
    TITLE
  </button>
  <button
    className="search-filter__button btn"
    onClick={[Function]}
    value="genres"
  >
    GENRE
  </button>
</div>
`;

exports[`search filter tests search filter test 2 1`] = `
<div
  className="search-filter"
>
  <span
    className="search-filter__text"
  >
    SEARCH BY
  </span>
  <button
    className="search-filter__button btn"
    onClick={[Function]}
    value="title"
  >
    TITLE
  </button>
  <button
    className="search-filter__button search-filter__button-active"
  >
    GENRE
  </button>
</div>
`;
